server/src/main/java/io/littlehorse/server/streams/util/AsyncWaiters.java

```diff
@@ -34,12 +35,12 @@ public class AsyncWaiters {
     private final HashMap<WfRunIdModel, GroupOfObserversWaitingForEvent> eventWaiters;
     private final Lock eventWaiterLock = new ReentrantLock();
 
-    public AsyncWaiters(ScheduledExecutorService executor) {
+    public AsyncWaiters() {
         commandWaiters = new ConcurrentHashMap<>();
         eventWaiters = new HashMap<>();
-
-        executor.scheduleAtFixedRate(this::cleanupOldCommandWaiters, 0, 10, TimeUnit.SECONDS);
-        executor.scheduleAtFixedRate(this::cleanupOldWorkflowEventWaiters, 1, 10, TimeUnit.SECONDS);
+        ScheduledExecutorService scheduledService = Executors.newScheduledThreadPool(2);
```

Comment:
Do we really want to create two full platform threads for this work? That seems pretty wasteful. Can we find another threadpool somewhere to use for it? (At the very least we should use just one thread)